### PR TITLE
Wallets now show proper jobs on Sechuds and Comms

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Wallets and PDAs now correctly display the job of the ID they have for both comms and sechud. Wallets with multiple IDs will display the first ID. 

PDAs without an ID inserted, will show their cached job (Owner: Blah-name name, **Head of Security** ). This cached job on PDAs is how it has worked for a long time, if this should be removed, I can do so.

This also eliminates the unused procs GetJobRealName and FindNameFromID, because they are terrible code.

It also moves GetIdCard to a carbon/human proc
         
## Why It's Good For The Game
Wallets not showing IDs is confusing for security, and showing the wrong ID on comms is also confusing.

## Images of changes

image
image

## Testing

No ID - shows no id on sechud, "No id" on comms
Wallet, HOS ID first - HOS sechud, HOS on comms
Wallet, Captain ID first - Captain sechud, Captain on comms
HOS PDA, no ID - HOS sechud, HOS on comms (note, this doesnt grant you HOS access, just makes you look like one)
HOS PDA, HOS ID in PDA - HOS sechud, HOS on comms
HOS PDA, Captain ID in PDA - Captain sechud, Captain on comms

## Changelog
:cl:
tweak: Wallets and PDAs will now consistently show the right ID on sechud and comms.
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
God bless, I can now carry cash around to pay the Barkeep/Chef with (and get robbed for).

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -818,3 +818,48 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /// Called on cyborg items that need special charging behavior. Override as needed for specific items.
 /obj/item/proc/cyborg_recharge(coeff = 1, emagged = FALSE)
 	return
+
+// Access and Job stuff
+
+/obj/item/proc/GetJobName() //Used in secHUD icon generation
```
Comment (Sirryan2002) [inline code/game/objects/items.dm:824]: could you convert this to `snake_case`

File: code/game/objects/items/weapons/storage/wallets.dm [hunk 2]
```diff
@@ -31,19 +31,32 @@
...
 
-/obj/item/storage/wallet/remove_from_storage(obj/item/W as obj, atom/new_location)
-	. = ..(W, new_location)
+/obj/item/storage/wallet/remove_from_storage(obj/item/I, atom/new_location)
+	. = ..()
 	if(.)
-		if(W == front_id)
-			front_id = null
-			update_icon(UPDATE_ICON_STATE)
+		if(istype(I, /obj/item/card/id))
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/storage/wallets.dm:37]: just do `if(. && istype(I, /obj/item/card/id))`
here

File: code/game/objects/items/weapons/storage/wallets.dm [hunk 3]
```diff
@@ -31,19 +31,32 @@
...
+			refresh_ID()
 
-/obj/item/storage/wallet/handle_item_insertion(obj/item/W as obj, prevent_warning = 0)
-	. = ..(W, prevent_warning)
+/obj/item/storage/wallet/handle_item_insertion(obj/item/I, prevent_warning = FALSE)
+	. = ..()
 	if(.)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/storage/wallets.dm:42]: same here

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -430,39 +430,45 @@
...
+	if(istype(id) && id.registered_name)
+		return id.registered_name
+
+	if(is_pda(wear_id))
+		var/obj/item/pda/pda = wear_id
+		return pda.owner
+
+	return if_no_id
+
+/mob/living/carbon/human/proc/GetIdCard(mob/living/carbon/human/H)
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/human.dm:465]: `snake_case`? 👉 👈

Comment (Contrabang) [reply code/modules/mob/living/carbon/human/human.dm]: snake_cased, but this also makes it a subtype of an already existing proc on mob called /get_id_card, so I just made this the human child proc of it

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/game/objects/items.dm [hunk 5]
```diff
@@ -818,3 +818,48 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
...
+	var/rankName = get_ID_rank(if_no_id = "Unknown")
+
+	var/job_icons = get_all_job_icons()
+	var/centcom = get_all_centcom_jobs()
+	var/solgov = get_all_solgov_jobs()
+
+	if(assignmentName in centcom) //Return with the NT logo if it is a Centcom job
+		return "Centcom"
+	if(rankName in centcom)
+		return "Centcom"
```
Comment (farie82) [inline code/game/objects/items.dm:835]:
```suggestion
	if((assignmentName in centcom) || (rankName in centcom)) //Return with the NT logo if it is a Centcom job
		return "Centcom"
```

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -818,3 +818,48 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
...
+
+	if(assignmentName in solgov) //Return with the SolGov logo if it is a SolGov job
+		return "solgov"
+	if(rankName in solgov)
+		return "solgov"
```
Comment (farie82) [inline code/game/objects/items.dm:840]:
```suggestion
	if((assignmentName in solgov) || (rankName in solgov)) //Return with the SolGov logo if it is a SolGov job
		return "solgov"
```

File: code/game/objects/items.dm [hunk 7]
```diff
@@ -818,3 +818,48 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
...
 /obj/item/proc/cyborg_recharge(coeff = 1, emagged = FALSE)
 	return
+
+// Access and Job stuff
+
+/obj/item/proc/get_job_name() //Used in secHUD icon generation
+	var/assignmentName = get_ID_assignment(if_no_id = "Unknown")
+	var/rankName = get_ID_rank(if_no_id = "Unknown")
+
+	var/job_icons = get_all_job_icons()
```
Comment (farie82) [inline code/game/objects/items.dm:828]:
```suggestion
	var/static/list/job_icons = get_all_job_icons()
```
In theory this one can change. But in practice it really shouldn't

File: code/game/objects/items.dm [hunk 8]
```diff
@@ -818,3 +818,48 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
...
+	var/centcom = get_all_centcom_jobs()
+	var/solgov = get_all_solgov_jobs()
```
Comment (farie82) [inline code/game/objects/items.dm:830]:
```suggestion
	var/static/list/centcom = get_all_centcom_jobs()
	var/static/list/solgov = get_all_solgov_jobs()
```

**Comment by SteelSlayer** [comment]
LGTM after Farie's changes.

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**